src/main/java/stats/utils/Cluster.java

```diff
@@ -0,0 +1,81 @@
+package stats.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cluster {
+
+    /**
+     * Constructor for new Cluster.
+     * @param id
+     */
+    public Cluster(int id) {
+        this.id = id;
+        this.points = new ArrayList();
+        this.centroid = null;
+    }
+
+
+    /**
+     * Clears the Cluster.
+     */
+    public void clear() {
+        points.clear();
+    }
+
+
+    /**
+     * Adds points to Cluster.
+     */
+    public void addPoint(Point point) {
+        points.add(point);
+    }
+
+
+    public List<Point> getPoints() {
+        return points;
+    }
+
+
+    public void setPoints(List points) {
+        this.points = points;
+    }
+
+
+    public Point getCentroid() {
+        return centroid;
+    }
+
+
+    public void setCentroid(Point centroid) {
+        this.centroid = centroid;
+    }
+
+
+    public int getId() {
+        return id;
+    }
+
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+
+    /**
+     * List of Points in Cluster.
+     */
+    private List<Point> points;
```

Comment:
Typically a data point is associated with a unique Id and if not is not too much to ask for this...so we only need to hold these ids here

Reply:
so, again I don't quite get what are you trying to say. remove `points`?

Reply:
Since you recalculate the cluster centroid outside of the class ```Cluster``` you just need a ```List<Integer>``` here